<?xml version="1.0"?>
<!-- vim: set fenc=utf-8 et sts=4 ft=xml : -->
<launch>

    <node name="fovis_visual_odometry"
        ns="/processes/odometry"
        pkg="fovis_ros"
        type="fovis_stereo_odometer"
        output="screen"
        >
        <remap from="stereo" to="/cameras/navigation/center"/>
        <remap from="image" to="/rect_mono"/>
        <param name="queue_size" type="int" value="5"/>
        <param name="approximate_sync" type="bool" value="true"/>
        <param name="odom_frame_id" type="string" value="visual_odom"/>
        <param name="base_link_frame_id" type="string" value="base_link"/>
        <param name="publish_tf" type="bool" value="false"/>
        <param name="bucket_height" type="string" value="80"/>
        <param name="bucket_width" type="string" value="80"/>
        <param name="clique_inlier_threshold" type="string" value="0.1"/>
        <param name="fast_threshold" type="string" value="20"/>
        <param name="fast_adaptive_gain" type="string" value="0.005"/>
        <param name="feature_search_window" type="string" value="100"/>
        <param name="feature_window_size" type="string" value="17"/>
        <param name="inlier_max_reprojection_error" type="string" value="5.0"/>
        <param name="max_keypoints_per_bucket" type="string" value="25"/>
        <param name="max_mean_reprojection_error" type="string" value="20.0"/>
        <param name="max_pyramid_level" type="string" value="3"/>
        <param name="min_features_for_estimate" type="string" value="10"/>
        <param name="min_pyramid_level" type="string" value="0"/>
        <param name="ref_frame_change_threshold" type="string" value="150"/>
        <param name="stereo_max_disparity" type="string" value="48"/>
        <param name="stereo_max_dist_epipolar_line" type="string" value="1.5"/>
        <param name="stereo_max_refinement_displacement" type="string" value="1.0"/>
        <param name="stereo_require_mutual_match" type="string" value="1"/>
        <param name="target_pixels_per_feature" type="string" value="250"/>
        <param name="update_target_features_with_refined" type="string" value="0"/>
        <param name="use_adaptive_threshold" type="string" value="1"/>
        <param name="use_bucketing" type="string" value="1"/>
        <param name="use_homography_initialization" type="string" value="1"/>
        <param name="use_image_normalization" type="string" value="1"/>
        <param name="use_subpixel_refinement" type="string" value="1"/>
        <!--
        -->
    </node>

</launch>
